design/experimental-new-features-in-modular.md

```diff
@@ -0,0 +1,204 @@
+# Shipping experimental new features in Modular
+
+## Background
+
+As the Azure SDK for JavaScript continues to evolve, it is important for us to have a dedicated space to ship experimental features within our own client or language. By providing a separate subpath export for experimental features alongside with the regular release of Modular, we can gather valuable feedback from our customers who have tried these new features. This approach enables us to receive early insights and iterate on the experimental features while ensuring that our mainline releases remain stable and reliable.
+
+## Questions
+
+1. **Version picking strategy**  
+  One latest GA + One latest preview version.  
+  If the latest preview version < latest GA version, then we will use latest GA version as the experiment version.  
+  _Questions_:
+    1. do we allow customers to set their own preview version?
```

Comment:
It's really risky to allow customers to specify alternative API versions than what the client declares support for. For example, if there's a private preview in circulation, it may have response features that could crash the client, falsify our types, or otherwise produce really bizarre results that our client code doesn't (can't) account for. I'm inclined to think that if a customer wants to use a custom API version then they probably need to make REST calls manually.

@bterlson @mpodwysocki wdyt?

Reply:
Fine with that.